apps/block_scout_web/assets/js/omni.js

```diff
@@ -0,0 +1,163 @@
+    "height: inherit;margin-left: 1vw;background-color: #8258cd;padding: 0 0.5vw;border-radius: 1vw;";
+  div.innerHTML =
+    '<img style="height: 1.5vw;display: inline-block;margin-top: -0.1vw" src="/images/favicon-32x32.png"> <div style="margin-top: 0.2vw;display: inline-block;font-weight: bolder;color: white;">X-Chain</div>';
+  return div;
+};
+
+const customXChainDataView = (details) => {
+  const div = document.createElement("div");
+  div.style =
+    "height: inherit;background-color: rgb(130, 88, 205);padding: 2.5vw 1.5vw 0vw 1.5vw;";
+  div.innerHTML = `
+  <div>
+  <img style="height: 2vw; display: inline-block;margin-top: -0.3vw" src="/images/favicon-32x32.png"> 
+  <div style="margin-top: 0.1vw;display: inline-block;font-weight: bolder;color: white;font-size: 1.2vw;padding-left: 0.5vw;">X-Chain Transaction Details</div></div>
+  <pre style="font-size: 1vw;display: block;margin-top: 0.5vw;font-weight: bolder;color: white;padding-left: 1vw;">
+  ${JSON.stringify(details, null, 4)}
+  </pre>`;
+  return div;
+};
+
+// expected return {txHash1:true, txHas2:true...}
+const checkTx = async (elements) => {
+  const response = await fetch(url + "/checkOmniTxHashes", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      txHashes: elements,
+    }),
+  });
+  if (response.status == 400) {
+    await response.json().then(console.error);
+    return undefined;
+  }
+  return response.json();
+};
+
+const getTxDetails = async (omniTxHash) => {
+  // dummy response
+  const response = await fetch(url + `/omniTx/${omniTxHash}`);
+  const data = await response.json();
+  if (data.length === 0) {
+    return undefined;
+  }
+  return data;
+};
+
+const getAllTxHashElements = () => {
+  return document.querySelectorAll('[data-test="transaction_hash_link"]');
+};
+
+const handletxHashLinks = async () => {
+  const elements = getAllTxHashElements();
+  if (elements.length == 0) {
+    return;
+  }
+  // check local cache
+  const unknownTxList = [];
+  for (let i of elements) {
+    if (i.parentNode.childElementCount <= 2 && txMap[i.innerHTML] === true) {
+      console.log("from cache " + i.innerHTML);
+      i.parentNode.appendChild(customMarker());
+    } else if (txMap[i.innerHTML] === undefined) {
+      unknownTxList.push(i.innerHTML);
+    }
+  }
+  if (unknownTxList.length == 0) {
+    return;
+  }
+  const xChainTxs = await checkTx(unknownTxList);
+  if (xChainTxs === undefined) {
+    return;
+  }
+  for (let i of elements) {
+    if (xChainTxs[i.innerHTML] === true) {
+      txMap[i.innerHTML] = true;
+      console.log("from api " + i.innerHTML);
+      i.parentNode.appendChild(customMarker());
+    } else if (txMap[i.innerHTML] == undefined) {
+      txMap[i.innerHTML] = false;
+    }
+  }
+};
+
+const displayTxDetails = async (txHash) => {
+  const elements = document.querySelectorAll(
+    '[class="card js-ad-dependant-mb-3"]'
+  );
+  if (elements.length != 1) {
```

Comment:
if we have control over this maybe we can add id and getElementById but this is also ok

Reply:
I don't think we can make that change without changing the server code for blockscout, this works for now until blockscout changes their ui components.